api/backup.go

```diff
@@ -122,8 +122,15 @@ func (s *Server) BackupDelete(w http.ResponseWriter, req *http.Request) error {
 
 	volName := mux.Vars(req)["volName"]
 
-	if err := s.m.DeleteBackup(input.Name, volName); err != nil {
-		return errors.Wrapf(err, "error deleting backup %v of volume %v", input.Name, volName)
+	backup, err := s.m.GetBackup(input.Name, volName)
+	if err != nil {
+		logrus.Errorf("Error getting backup %v of volume %v: %v", input.Name, volName, err)
+	}
+
+	if backup != nil {
+		if err := s.m.DeleteBackup(input.Name, volName); err != nil {
```

Comment:
The fix is ok, but I feel we can fix it general for all resource deletion if the target is not found. 

Do you know what error will be returned?

```go
s.lhClient.LonghornV1beta2().Backups(s.namespace).Delete(context.TODO(), backupName, metav1.DeleteOptions{})
```

Reply:
After discussion, because the return of the Delete function is the implementation of Error(). 

It's hard to distinguish error classes without getting the HTTP status code.

https://github.com/longhorn/longhorn-manager/blob/d4d5f8943f62cfe84c4745fd93a796c3e0789033/k8s/pkg/client/clientset/versioned/typed/longhorn/v1beta2/backup.go#L156-L164